Sources/FigmaExportCore/Processor/AssetsProcessor.swift

```diff
@@ -270,30 +299,40 @@ public extension AssetsProcessable {
             .filter { $0.platform == platform || $0.platform == nil }
             .sorted { $0.name < $1.name }
 
-        let darkColors = darkSet
-            .filter { $0.platform == platform || $0.platform == nil }
-            .sorted { $0.name < $1.name }
+        // After validations, only those dark colors in the light color palette are allowed
+        // However the dark array may be smaller than the light array
+        // Create a same size array of dark colors so we can zip in the next step
+        let darkColorMap: [String: AssetType] = darkSet.reduce(into: [:]) { $0[$1.name] = $1 }
+        let darkColors = lightColors
+            .map { lightColor in darkColorMap[lightColor.name] }
 
         let zipResult = zip(lightColors, darkColors)
 
         return zipResult
             .map { lightAsset, darkAsset in
+                if let darkAsset = darkAsset {
+                    return AssetPair(light: processedAssetName(lightAsset), dark: processedAssetName(darkAsset))
+                } else {
+                    return AssetPair(light: processedAssetName(lightAsset), dark: nil)
```

Comment:
Get rid of code duplication by using `map`:
```swift
AssetPair(
    light: processedAssetName(lightAsset),
    dark: darkAsset.map { processedAssetName($0) }
)
```

Reply:
Much cleaner. Thanks.